### ISSUE TITLE
RDLibrary: Segfault when editing markers on imported cut on a cart.

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#799]
Hi.

I'm starting to experiment with several cuts for a same Cart.
The Idea is to have a Radio Edit cut, toghether with the original extended/club Mix on the same cart, so, 'daytime' airing could be more 'comercial' while night time be more 'club-style' ... but propper markers have to be setup anyways for smooth transitioning ...and that's how I've found a consistent segfault:

I import a second cut into a cart, and then, I edit the first cart markers... I end the edit so I return to the cart screen, then I select second cut and click edit markers again: RDLibrary segfaults (I'll bet is qt5 mess with listing items or the like)

Here's GDB backtrace:

```
Thread 1 "rdlibrary" received signal SIGSEGV, Segmentation fault.
0x00007ffff655e9bb in QPen::operator==(QPen const&) const () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
(gdb) backtrace
#0  0x00007ffff655e9bb in QPen::operator==(QPen const&) const ()
   from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#1  0x00007ffff6c3dc30 in QGraphicsLineItem::setPen(QPen const&) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#2  0x00007ffff7948ae7 in RDMarkerView::WriteWave (this=0x555555c082e0)
    at rdmarkerview.cpp:1272
#3  0x00007ffff7948f0b in RDMarkerView::setCut (this=0x555555c082e0, 
    err_msg=0x7fffffffc6e8, cartnum=10689, cutnum=1) at rdmarkerview.cpp:692
#4  0x00007ffff793d994 in RDMarkerDialog::exec (this=0x555555c07ac0, 
    cartnum=10689, cutnum=1) at rdmarkerdialog.cpp:155
#5  0x000055555557858b in AudioCart::editCutData() ()
#6  0x00007ffff5cf75e0 in ?? () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#7  0x00007ffff6a36e42 in QAbstractButton::clicked(bool) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#8  0x00007ffff6a377aa in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#9  0x00007ffff6a390ff in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#10 0x00007ffff6a392d1 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#11 0x00007ffff6989fae in QWidget::event(QEvent*) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#12 0x00007ffff694a15f in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#13 0x00007ffff695104b in QApplication::notify(QObject*, QEvent*) ()
--Type <RET> for more, q to quit, c to continue without paging--c
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#14 0x00007ffff5cc0fca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#15 0x00007ffff6950076 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#16 0x00007ffff69a2011 in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#17 0x00007ffff69a533e in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#18 0x00007ffff694a15f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#19 0x00007ffff5cc0fca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#20 0x00007ffff6262583 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#21 0x00007ffff623864c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#22 0x00007ffff15cd7da in ?? () from /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#23 0x00007ffff43dbe6b in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007ffff43dc118 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#25 0x00007ffff43dc1cf in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#26 0x00007ffff5d1851f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#27 0x00007ffff5cbf98b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#28 0x00007ffff6b42e77 in QDialog::exec() () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#29 0x000055555559897d in MainWidget::editData() ()
#30 0x00007ffff5cf75e0 in ?? () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#31 0x00007ffff6a36e42 in QAbstractButton::clicked(bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#32 0x00007ffff6a377aa in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#33 0x00007ffff6a390ff in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#34 0x00007ffff6a392d1 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#35 0x00007ffff6989fae in QWidget::event(QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#36 0x00007ffff694a15f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#37 0x00007ffff695104b in QApplication::notify(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#38 0x00007ffff5cc0fca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#39 0x00007ffff6950076 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#40 0x00007ffff69a2011 in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#41 0x00007ffff69a533e in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#42 0x00007ffff694a15f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#43 0x00007ffff5cc0fca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#44 0x00007ffff6262583 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#45 0x00007ffff623864c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#46 0x00007ffff15cd7da in ?? () from /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#47 0x00007ffff43dbe6b in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#48 0x00007ffff43dc118 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#49 0x00007ffff43dc1cf in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#50 0x00007ffff5d1851f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#51 0x00007ffff5cbf98b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#52 0x00007ffff5cc7c00 in QCoreApplication::exec() () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#53 0x0000555555575f4d in main ()
```

EDIT:

Got RDLibrary to crash after trying to edit a freshly imported cut: I create a new cart, and then import from file audio to its single/initial cut, close the cut window, then again in cart window, click edit markers... crash.
Not sure if it is the same crash, but I feel it's all about frontend/qt5 status not coping with backend changes. 

Regards!

**Comment by fpeyraud** [comment]
Hello,
I confirm the bug on commit 037b8b9fe54348874f7c662be30f2d1824acb9a4 when trying to edit the markers of the second cut of a musical cart. I had the crash also when trying  to edit the markers of a single-cut cart twice in a row without closing the cart details window.

**Comment by alexolivan** [comment]
Hi again.

I've also realized that, if editing a cart, one opens the scheduler codes menu and presses Save (I guess that triggers some backend stuff) and then afterwards tries to edit cut markers, then RDLibrary crashes too.
However, if the scheduler codes are deployed and dismissed without saving, no problem occurs... feels like any interaction frontend-backend prior to edit markers makes RDLibrary to crash.

**Comment by ElvishArtisan** [comment]
Confirmed here.

**Comment by ElvishArtisan** [comment]
Fixed in 23fcc6e4. Please test!

**Comment by alexolivan** [comment]
Unable to reproduce it anymore :-) ... closing!
